app/src/main/java/org/dhis2/utils/DatePickerUtils.java

```diff
@@ -64,12 +107,21 @@ private static Dialog buildDialog(Context context,
                                       @Nullable Date minDate,
                                       @Nullable Date maxDate,
                                       boolean allowFutureDates,
+                                      boolean fromOtherPeriods,
                                       OnDatePickerClickListener buttonListener) {
         LayoutInflater layoutInflater = LayoutInflater.from(context);
         WidgetDatepickerBinding widgetBinding = WidgetDatepickerBinding.inflate(layoutInflater);
         final DatePicker datePicker = widgetBinding.widgetDatepicker;
         final DatePicker calendarPicker = widgetBinding.widgetDatepickerCalendar;
 
+        preferences = ((App) datePicker.getContext().getApplicationContext()).appComponent().preferenceProvider();
```

Comment:
I do not like this, but if anyone has a better idea to 'inject' the PreferenceProvider in this class, let me know.

Reply:
Maybe DatePicker utils has too many responsibilities, if the purpose of this is to populate a Dialog, I would consider to refactor this class into a **customview** and maybe this **customview** can contain a **presenter** or **viewmodel** in order to process the logic, then you can inject preferences into the presenter o ViewModel. Also I would consider as well to move this to **commons** module. What do you think?